### DISCUSSION TITLE
Missing settings in dconf?

### DISCUSSION THREAD

**Discussion by peteruithoven** [elementary/gala#1444] [Q&A] [answered]
To debug an issue I wanted to disable `move-fullscreened-workspace` defined here:
https://github.com/elementary/gala/blob/202755c38afccc9cf4fbfbdc62a093541987747f/data/gala.gschema.xml#L103-L107
It should be available under the path `/org/pantheon/desktop/gala/behavior/`. But for some reason it's not:
```
$ dconf list /org/pantheon/desktop/gala/behavior/
hotcorner-bottomleft
hotcorner-bottomright
hotcorner-topleft
hotcorner-topright
overlay-action
```
Same when I navigate there using the dconf-editor:
![Screenshot from 2022-07-08 16-12-23@2x](https://user-images.githubusercontent.com/523210/178009414-977ad840-f13b-4b10-902c-e05fd72443e9.png)

There is no clear difference between the settings that are listed and the ones that are not? What am I missing here?

**Comment by jeremypw** [comment]
I get the same list as you with `dconf list` but I can see all the keys with dconf-editor.  The ones visible with list are **bold** in the editor.  The only difference I could see is that the bold entries were _not_ the default value.  If you set a default value to a non-default, then re-run `dconf list` the changed setting appears in the list. This is on ElementaryOS7

**Reply by peteruithoven** [reply]
This is super weird. I'm not finding a relevant setting that would allow me to show all vs only modified settings in the dconf-editor. 

Just to confirm I've added a screenshot. 

Also weird, when I try to read the current or default value I get nothing:
```
$ dconf read /org/pantheon/desktop/gala/behavior/move-fullscreened-workspace
$ dconf read -d /org/pantheon/desktop/gala/behavior/move-fullscreened-workspace
```
When I try the same with `hotcorner-bottomleft` it does work:
```
$ dconf read /org/pantheon/desktop/gala/behavior/hotcorner-bottomleft 
'none'
```

Also note that when a key doesn't exist there is no feedback, no error.

**Reply by jeremypw** [reply]
My dconf-editor is Version: 3.38.3-3 (native install) - I have not changed any settings.  I recently installed Gala from source.

This is what I get (dconf-editor looks a bit different from yours):

![Screenshot from 2022-07-08 16 47 02](https://user-images.githubusercontent.com/10513844/178026388-1ea23b0a-1c47-44d8-b1e8-3a9bc0955e1e.png)

**Reply by jeremypw** [reply]
Have you tried `gsettings list-keys org.pantheon.desktop.gala.behavior` - this lists all the keys for me?

**Reply by peteruithoven** [reply]
Ah interesting, gsettings does indeed list all.
```
$ gsettings list-keys org.pantheon.desktop.gala.behavior
panel-main-menu-action
window-overview-type
dock-names
move-maximized-workspace
dynamic-workspaces
move-fullscreened-workspace
overlay-action
toggle-recording-action
hotcorner-topleft
hotcorner-custom-command
hotcorner-topright
hotcorner-bottomright
edge-tiling
use-new-notifications
hotcorner-bottomleft
```

**Reply by peteruithoven** [reply]
I'm running dconf-editor 3.36.7, installed through flatpak... Ah! That's the problem. After installing it through apt I do see all keys.
Mystery solved.

**Comment by jeremypw** [comment]
This behaviour does not appear to be documented in `man dconf` :shrug:

**Comment by peteruithoven** [comment, marked as answer]
Trick was installing dconf-editor natively, using `apt` instead of as Flatpak. Now I do see all the settings. 

Thanks @jeremypw for the help.

**Reply by jeremypw** [reply]
Yeah, presumably the sandbox does not have large enough holes punched in it in the Flatpak version of dconf-editor!
